crates/cubecl-core/src/runtime_tests/enums.rs

```diff
@@ -0,0 +1,153 @@
+        TestEnum::D => TestEnum::new_D(),
+        TestEnum::E { x } => TestEnum::new_E(x),
+        TestEnum::F { x, .. } => TestEnum::new_F(x, 2),
+    };
+}
+
+#[cube(launch)]
+pub fn kernel_scalar_enum(test: TestEnum<i32>, output: &mut Array<f32>) {
+    match test {
+        TestEnum::A(x, y) | TestEnum::F { x, y } => {
+            output[0] = f32::cast_from(x) + f32::cast_from(y);
+        }
+        TestEnum::B(b) => {
+            output[0] = f32::cast_from(b.x) + f32::cast_from(b.y);
+        }
+        TestEnum::C(x) | TestEnum::E { x } => {
+            output[0] = f32::cast_from(x);
+        }
+        TestEnum::D => {
+            output[0] = 999.0;
+        }
+    };
+}
+
+pub fn test_scalar_enum<R: Runtime>(client: ComputeClient<R::Server, R::Channel>) {
+    let array = client.empty(std::mem::size_of::<f32>());
+
+    kernel_scalar_enum::launch::<R>(
+        &client,
+        CubeCount::new_single(),
+        CubeDim::new_single(),
+        TestEnumArgs::<i32, R>::C(ScalarArg::new(10)),
+        unsafe { ArrayArg::<R>::from_raw_parts::<f32>(&array, 1, 1) },
+    );
+    let bytes = client.read_one(array.binding());
+    let actual = f32::from_bytes(&bytes);
+
+    assert_eq!(actual[0], 10.0);
+}
+
+#[derive(CubeLaunch)]
+pub enum ArrayFloatInt {
+    Float(Array<f32>),
+    Int(Array<i32>),
+}
+
+#[cube(launch)]
+fn kernel_array_float_int(array: &mut ArrayFloatInt) {
+    if UNIT_POS == 0 {
+        match array {
+            ArrayFloatInt::Float(array) => array[0] = 10.0,
+            ArrayFloatInt::Int(array) => array[0] = 20,
+        }
+    }
+}
+
+pub fn test_array_float_int<R: Runtime, T: CubePrimitive + CubeElement>(
+    client: &ComputeClient<R::Server, R::Channel>,
+    expected: T,
+) {
+    let array = client.empty(std::mem::size_of::<T>());
+
+    kernel_array_float_int::launch(
+        client,
+        CubeCount::new_single(),
+        CubeDim::new_single(),
+        if std::any::TypeId::of::<T>() == std::any::TypeId::of::<f32>() {
+            ArrayFloatIntArgs::Float(unsafe { ArrayArg::<R>::from_raw_parts::<f32>(&array, 1, 1) })
+        } else {
+            ArrayFloatIntArgs::Int(unsafe { ArrayArg::<R>::from_raw_parts::<i32>(&array, 1, 1) })
+        },
+    );
+
+    let bytes = client.read_one(array.binding());
+    let actual = T::from_bytes(&bytes);
+
+    assert_eq!(actual[0], expected);
+}
+
+#[allow(missing_docs)]
+#[macro_export]
+macro_rules! testgen_enums {
+    () => {
+        mod enums {
+            use super::*;
+
+            #[test]
+            fn scalar_enum() {
+                let client = TestRuntime::client(&Default::default());
+                cubecl_core::runtime_tests::enums::test_scalar_enum::<TestRuntime>(client);
+            }
+
+            #[test]
+            fn array_float_int() {
+                let client = TestRuntime::client(&Default::default());
+                cubecl_core::runtime_tests::enums::test_array_float_int::<TestRuntime, f32>(
```

Comment:
What does float_int mean?

Reply:
The enum as two variants `Float` and `Int`